internal/token/fileset.go

```diff
@@ -1,55 +1,82 @@
 package token
 
 import (
-	"go/token"
-
 	"github.com/influxdata/flux/ast"
 )
 
 type File struct {
-	file *token.File
+	name  string
+	lines []int
+	sz    int
 }
 
-func (f *File) AddLine(offset int) {
-	f.file.AddLine(offset)
+func NewFile(name string, sz int) *File {
+	return &File{
+		name:  name,
+		lines: []int{0},
+		sz:    sz,
+	}
 }
 
-func (f *File) Base() int {
-	return f.file.Base()
+func (f *File) AddLine(offset int) {
+	f.lines = append(f.lines, offset)
 }
 
 func (f *File) Name() string {
-	return f.file.Name()
+	return f.name
 }
 
-func (f *File) Size() int {
-	return f.file.Size()
+// Offset returns the offset for the given line/column.
+func (f *File) Offset(pos ast.Position) int {
+	if pos.Line == 0 || pos.Column == 0 {
+		return -1
+	}
+	offset := f.lines[pos.Line-1]
+	return offset + pos.Column - 1
 }
 
-func (f *File) Pos(offset int) Pos {
-	return Pos(f.file.Pos(offset))
+func (f *File) Base() int {
+	return 1
```

Comment:
It indicates the beginning position of the file. It's taken from the go library and seems to mostly exist so `token.Pos = 0` is equal to `NoPos`. It makes more sense for the `token.FileSet` where you have multiple files in the same area and the base can be something other than 1.

It may be good to remove this in the future, but I do think that having `0` as an invalid value for `token.Pos` is good so I kept it.